routing_common/transit_serdes.hpp

```diff
@@ -219,5 +219,45 @@ class Deserializer
 private:
   Source & m_source;
 };
+
+template <typename Sink>
+class FixedSizeSerializer
+{
+public:
+  explicit FixedSizeSerializer(Sink & sink) : m_sink(sink) {}
+
+  template <typename T>
+  typename std::enable_if<std::is_integral<T>::value || std::is_enum<T>::value, void>::type
+  operator()(T const & t, char const * /* name */ = nullptr)
+  {
+    WriteToSink(m_sink, t);
+  }
+
```

Comment:
Why not to:
```c++
template <typename T>
struct IsIntegralOrEnum {
   static int const kValue = std::is_integral<T>::value || std::is_enum<T>::value;
};

class FixedSizeSerializer
{
public:
  ...
  template <typename T>
  typename std::enable_if<IsIntegralOrEnum<T>::kValue>::type operator()(...) {}
  
  template<typename T>
  typename std::enable_if<!IsIntegralOrEnum<T>::kValue>::type operator()(T const & t, char const * /* name */ = nullptr)
  {
    t.Visit(*this);
  }
  ...
};
```
This will allow you to get rid of VisitHeader.

Reply:
Yes. It's true. But on the other hand I get a complication in FixedSizeSerializer/FixedSizeDeserializer. In that class these classes became ready for recursive. Are you sure that adding operator()(T &) to these classes is better then a simple method VisitHeader()?

Reply:
What's the problem with recursion? It will allow you to re-use this classes for any hierarchy consisting of structs and integrals. But, if you're scared by recursion, why not to:
```c++
void operator()(TransitHeader & header, char const * /* name */ = nullptr)
{
   header.Visit(*this);
}
```
Almost the same, as VisitHeader, but:
* VisitHeader claims that only header may be visited, but the code does not depend on this fact and actually can work with any class.
* header is non-const in VisitHeader, this is wrong - for serializing I should pass constant header.

Reply:
+1, good point

Reply:
I added `void operator()(TransitHeader const & header) { header.Visit(*this); }`/`void operator()(TransitHeader & header) { header.Visit(*this); }` to `FixedSizeSerializer` and `FixedSizeDeserializer` classes. And remove VisitHeader() method.